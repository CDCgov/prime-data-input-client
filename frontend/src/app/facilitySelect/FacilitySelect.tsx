import React from 'react';

import Button from '../commonComponents/Button';
import { formatFullName } from '../utils/user';

import './FacilitySelect.scss';

interface Props {
  setActiveFacility: (facility: Facility) => void;
  facilities: Facility[];
  organization: Organization;
  user: User;
}

const FacilitySelect: React.FC<Props> = ({
  facilities,
  organization,
  user,
  setActiveFacility,
}) => {
  return (
    <main className="prime-home" id="facility-select">
      <div className="grid-container">
        <div className="grid-row position-relative">
          <div className="prime-container usa-card__container">
            <div className="usa-card__header">
              <div>
                <h2>SimpleReport</h2>
                <div className="organization-name">{organization.name}</div>
              </div>
            </div>
            <div className="usa-card__body">
              <p className="welcome">Welcome, {formatFullName(user)}</p>
              <p className="select-text">
                Please select which facility you are working at today
              </p>
              {facilities.map((f) => (
                <Button
                  key={f.id}
                  onClick={() => setActiveFacility(f)}
                  variant="outline"
                >
                  {f.name}
                </Button>
              ))}
            </div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default FacilitySelect;
