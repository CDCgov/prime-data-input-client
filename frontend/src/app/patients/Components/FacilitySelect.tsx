import React from "react";
import { useTranslation } from "react-i18next";
import { useSelector } from "react-redux";

import Select from "../../commonComponents/Select";
import { RootState } from "../../store";

interface Props {
  facilityId: string | null;
  onChange: (facilityId: string | null) => void;
  validateField: () => void;
  validationStatus: (name: keyof PersonFormData) => "error" | undefined;
  errors: { [key: string]: string | undefined };
  hidden?: boolean;
}

const ALL_FACILITIES = "~~ALL-FACILITIES~~";
const NAME = "facilityId";

const FacilitySelect: React.FC<Props> = (props) => {
  const { t } = useTranslation();

  const facilities = useSelector<RootState, Facility[]>(
    (state) => state.facilities
  );

  if (props.hidden) {
    return null;
  }

  const facilityList = facilities.map((f) => ({
    label: f.name,
    value: f.id,
  }));
  facilityList.unshift({
    label: t("facility.form.allFacilities"),
    value: ALL_FACILITIES,
  });

  const onChange = (value: string | null) => {
    props.onChange(value === ALL_FACILITIES ? null : value);
  };

  return (
    <Select
      label={t("facility.form.heading")}
      name={NAME}
      value={props.facilityId === null ? ALL_FACILITIES : props.facilityId}
      onChange={onChange}
      onBlur={props.validateField}
      validationStatus={props.validationStatus(NAME)}
      errorMessage={props.errors[NAME]}
      options={facilityList}
      defaultOption={t("common.defaultDropdownOption")}
      defaultSelect={true}
      required
    />
  );
};

export default FacilitySelect;
