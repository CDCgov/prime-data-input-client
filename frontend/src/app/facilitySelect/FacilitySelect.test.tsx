import React from "react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";

import FacilitySelect from "./FacilitySelect";
import { Provider } from "react-redux";

const mockStore = configureStore([]);

describe("FacilitySelect", () => {
  let component: any;
  const mockSetActiveFacility = jest.fn();

  beforeEach(() => {
    const store = mockStore({
      organization: {
        name: "Organization Name",
      },
      user: {
        firstName: "Kim",
        lastName: "Mendoza",
      },
      facilities: [
        { id: "1", name: "Facility 1" },
        { id: "2", name: "Facility 2" },
      ],
    });

    component = renderer.create(
      <Provider store={store}>
        <FacilitySelect
          facilities={
            [
              { id: "1", name: "Facility 1" },
              { id: "2", name: "Facility 2" },
            ] as any
          }
          setActiveFacility={mockSetActiveFacility}
        />
      </Provider>
    );
  });

  it("should render with a value", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("On facility select", () => {
    beforeEach(() => {
      renderer.act(() => {
        component.root.findAllByType("button")[0].props.onClick();
      });
    });
    it("should call setActiveFacility once", () => {
      expect(mockSetActiveFacility).toHaveBeenCalledTimes(1);
    });
    it("should select the first facility", () => {
      expect(mockSetActiveFacility).toHaveBeenCalledWith({
        id: "1",
        name: "Facility 1",
      });
    });
  });
});
