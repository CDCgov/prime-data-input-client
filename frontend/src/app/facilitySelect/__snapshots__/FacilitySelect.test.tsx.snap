// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilitySelect should render with a value 1`] = `
<main
  className="prime-home"
  id="facility-select"
>
  <div
    className="grid-container"
  >
    <div
      className="grid-row position-relative"
    >
      <div
        className="prime-container card-container"
      >
        <div
          className="usa-card__header"
        >
          <div>
            <h2>
              SimpleReport
            </h2>
            <div
              className="organization-name"
            >
              Organization Name
            </div>
          </div>
        </div>
        <div
          className="usa-card__body"
        >
          <p
            className="welcome"
          >
            Welcome, 
            Kim Mendoza
          </p>
          <p
            className="select-text"
          >
            Please select the testing facility where you are working today.
          </p>
          <button
            className="usa-button usa-button--outline"
            onClick={[Function]}
            type="button"
          >
            Facility 1
          </button>
          <button
            className="usa-button usa-button--outline"
            onClick={[Function]}
            type="button"
          >
            Facility 2
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;
